custom_components/magiqtouch/climate.py

```diff
@@ -146,20 +199,63 @@ def target_temperature(self):
         """Return the temperature we try to reach."""
         return self.controller.current_state.CTemp
 
+    @property
+    def max_temp(self):
+        return self.controller.get_installed_device_config().get("MaximumTemperature", 35)
+
+    @property
+    def min_temp(self):
+        return self.controller.get_installed_device_config().get("MinimumTemperature", 7)
+
+    @property
+    def hvac_action(self):
+        on = self.controller.current_state.SystemOn
+        if not on:
+            _LOGGER.debug(CURRENT_HVAC_OFF)
+            return CURRENT_HVAC_OFF
+        # Show as off if individual zone is off
+        zone_on = getattr(self.controller.current_state, self.controller.get_on_off_zone_name(self.zone_index))
+        if not self.controls_system and not zone_on:
+            _LOGGER.debug(CURRENT_HVAC_OFF)
+            return CURRENT_HVAC_OFF
+
+        if self.controller.current_state.HActualGasRate > 0 \
+            and self.controller.current_state.HActualFanSpeed > 0:
+            return CURRENT_HVAC_HEAT
+        elif self.controller.current_state.FAOCActualCompressorON == 1 \
+            or self.controller.current_state.IAOCCompressorON == 1 \
+            or self.controller.current_state.EvapCRunning == 1:
+            return CURRENT_HVAC_COOL
+        elif (self.controller.current_state.HFanOnly == 1 \
+                and self.controller.current_state.HActualFanSpeed > 0 \
+                and self.controller.current_state.HFanSpeed > 0) \
+            or self.controller.current_state.CFanOnlyOrCool == 1:
+            return CURRENT_HVAC_FAN
```

Comment:
Just an FYI the logic here isn't quite right, if breaks fan-only mode for me with my evaporative cooler. I'm not surprised though, the current_settings for evap/fan are pretty strange. I'll fix it in a follow up commit that hopefully shouldn't break heat/cool functionality for you & others!

Reply:
I had a feeling I was somehow going to break evap coolers, so cheers for pointing this out.